packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8sResource.js

```diff
@@ -0,0 +1,187 @@
+        this.resource = this.templateGenerator(this.templateConfig);
+
+        // Apply job `targets` setting as k8s nodeAffinity
+        // We assume that multiple targets require both to match ...
+        // NOTE: If you specify multiple `matchExpressions` associated with
+        // `nodeSelectorTerms`, then the pod can be scheduled onto a node
+        // only if *all* `matchExpressions` can be satisfied.
+        this._setAffinity();
+        this._setResources();
+        this._setVolumes();
+        this._setAssetsVolume();
+        this._setImagePullSecret();
+    }
+
+    _makeConfig() {
+        const clusterName = _.get(this.terasliceConfig, 'name');
+        const clusterNameLabel = clusterName.replace(/[^a-zA-Z0-9_\-.]/g, '_').substring(0, 63);
+        const configMapName = _.get(
+            this.terasliceConfig,
+            'kubernetes_config_map_name',
+            `${this.terasliceConfig.name}-worker`
+        );
+        const jobNameLabel = this.execution.name.replace(/[^a-zA-Z0-9_\-.]/g, '_').substring(0, 63);
+        const name = `ts-${this.nameInfix}-${jobNameLabel.substring(0, 42)}-${this.execution.job_id.substring(0, 13)}`;
+        const shutdownTimeoutMs = _.get(this.terasliceConfig, 'shutdown_timeout', 60000);
+        const shutdownTimeoutSeconds = Math.round(shutdownTimeoutMs / 1000);
+
+
+        const config = {
+            // assetsDirectory: _.get(this.terasliceConfig, 'assets_directory', ''),
+            // assetsVolume: _.get(this.terasliceConfig, 'assets_volume', ''),
+            clusterName,
+            clusterNameLabel,
+            configMapName,
+            dockerImage: _.get(this.terasliceConfig, 'kubernetes_image', 'teraslice:k8sdev'),
+            execution: safeEncode(this.execution),
+            exId: this.execution.ex_id,
+            jobId: this.execution.job_id,
+            jobNameLabel,
+            name,
+            namespace: _.get(this.terasliceConfig, 'kubernetes_namespace', 'default'),
+            nodeType: this.nodeType,
+            replicas: this.execution.workers,
+            shutdownTimeout: shutdownTimeoutSeconds
+        };
+        return config;
+    }
+
+    _makeTemplate(folder, fileName) {
+        const filePath = path.join(__dirname, folder, `${fileName}.hbs`);
+        const templateData = fs.readFileSync(filePath, 'utf-8');
+        const templateKeys = ['{{', '}}'];
+
+        return (config) => {
+            const templated = barbe(templateData, templateKeys, config);
+            return JSON.parse(templated);
+        };
+    }
+
+    _setImagePullSecret() {
+        if (this.terasliceConfig.kubernetes_image_pull_secret !== '') {
+            this.resource.spec.template.spec.imagePullSecrets = [
+                { name: this.terasliceConfig.kubernetes_image_pull_secret }
+            ];
+        }
+    }
+
+    _setAssetsVolume() {
+        if ((this.terasliceConfig.assets_directory !== '') && (this.terasliceConfig.assets_volume !== '')) {
```

Comment:
This can be simplified to:

```suggestion
        if (this.terasliceConfig.assets_directory && this.terasliceConfig.assets_volume) {
```